### PR TITLE
Add diagrams, and extend summaries

### PR DESCRIPTION
**Pull request by szszszsz**
- [x] One sentence: smart cards for the Web
- [x] Benefits for the users 
- [x] summary of the use cases
- [x] architecture / diagrams
  - [x] where openpgpjs is here
  - [x] how components are connected
  - [x] alice-bob communication, standardized approach
  - [x] diagrams for use cases
- [ ] check documentation for developers
- [x] change/typo: periodic -> timeout, auto-logout like
- [ ] describe better the initialization command, and setup as a whole
- [x] idea: allowed domains for the pan-wide resident keys, per RK
- [x] rename: webcrypt -> WebSmartCard

Make sure to view the diff in the richtext mode:
- https://github.com/Nitrokey/nitrokey-webcrypt/pull/28/files?short_path=b335630#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -50,30 +63,116 @@ Please refer to these defined terms throughout the documentation to ensure a con
 
 ## Implementation Details
 
-The implementation of Nitrokey WebCrypt incorporates the following design considerations:
+The implementation of Nitrokey WebSmartCard incorporates the following design considerations:
 
-1. **Inspiration from OnlyKey's WebCrypt**: The solution draws inspiration from OnlyKey's WebCrypt proof-of-concept,
+1. **Inspiration from OnlyKey's WebSmartCard**: The solution draws inspiration from OnlyKey's WebSmartCard proof-of-concept,
```
Comment (sosthene-nitrokey) [inline README.md]: I don't think this should be renamed here.

**Review by sosthene-nitrokey — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (szszszsz) [reply README.md]: Good catch, corrected

**Review by sosthene-nitrokey — COMMENTED** [review]
In the diagrams the "Nitrokey WebSmartCard API" would be loaded as a Javascript library correct?

I think it might be worth nesting it inside of the OpenPGP.js client application somehow to make it clearer what happens in the website, the browser, and the NK3

**Comment by szszszsz** [comment]
> In the diagrams the "Nitrokey WebSmartCard API" would be loaded as a Javascript library correct?

That's correct. It will be then injected as a plugin object into OpenPGP.js through its plugin mechanism.

> I think it might be worth nesting it inside of the OpenPGP.js client application somehow to make it clearer what happens in the website, the browser, and the NK3

I think this is a task for the communication channel diagram, and its description. I will see if I can improve it. One thing is to name `Nitrokey WebSmartCard API` as a Javascript library.
In general, while I agree adding server/service as another participant here is technically more precise, I am worried it will make the chart obscure.

**Review by sosthene-nitrokey — APPROVED** [review]

**Review by robin-nitrokey — APPROVED** [review]

**Comment by szszszsz** [comment]
@daringer Let's discuss later if, and what are you missing from the `Seed and Key Generation` chapter regarding initialization.
@sosthene-nitrokey I've boxed the Javascript app into a "Web Browser" group, and introduced an optional Web Service communication. Hopefully this clears the general image a bit.

**Pull request merged**